pkg/apis/serving/v1alpha2/inferenceservice_validation.go

```diff
@@ -121,3 +128,10 @@ func validateCanaryTrafficPercent(spec InferenceServiceSpec) error {
 	}
 	return nil
 }
+
+func validateInferenceServiceName(isvc *InferenceService) error {
```

Comment:
suggest checking the logic here in kubernetes
https://github.com/kubernetes/kubernetes/blob/e077b0ffa48b1c0300a2cb25dd5d45d6fc358ade/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L187

Reply:
Thanks, it has helped me a lot.

Reply:
Cool! can you help add a few unit tests for the name check?

Reply:
Sure, I added them